### DISCUSSION TITLE
vector of impulse response

### DISCUSSION THREAD

**Discussion by RFStudent74** [scikit-rf/scikit-rf#1091] [Q&A] [answered]
Hi, 
Is it possible to obtain and save in text format the vector of values ​​displayed by the plt_s_db_time() function? The time axis vector too?
Or, are there other functions that give the same result as plt_s_db_time()? Because I need the values ​​of the impulse response from the S21 matrix in text format?
Thanks in advance.

**Comment by FranzForstmayr** [comment, marked as answer]
Hi, 
this is basically the code to get the csv file.

```python
import skrf as rf
import pandas as pd

net = rf.data.ring_slot

db_time = pd.DataFrame({"time": net.frequency.t, "s_db": net.s21.s_time_db.flatten()})
db_time.to_csv("test.csv", index=None)

net.plot_s_db_time()
```
`plot_s_db_time` calls `plot_s_time_db` under the hood after applying a filter.

```python
def plot_s_db_time(netw: Network, *args, window: str | float | tuple[str, float]=('kaiser', 6),
        normalize: bool = True, center_to_dc: bool = None, **kwargs):
    return netw.windowed(window, normalize, center_to_dc).plot_s_time_db(*args,**kwargs)
```
`plot_s_time_db` and friends are autogenerated and can be called without `plot` as well, so `network.s_time_db`.

**Reply by RFStudent74** [reply]
Thank you very much for your help.
